examples/shader/compute_shader_game_of_life.rs

```diff
@@ -94,29 +121,93 @@ impl Plugin for GameOfLifeComputePlugin {
     }
 }
 
-#[derive(Resource, Clone, Deref, ExtractResource, AsBindGroup)]
-struct GameOfLifeImage {
-    #[storage_texture(0, image_format = R32Float, access = ReadWrite)]
-    texture: Handle<Image>,
+#[derive(Resource, Clone, ExtractResource)]
+struct GameOfLifeImages {
+    texture_a: Handle<Image>,
+    texture_b: Handle<Image>,
+}
+
+// Manual implementation of AsBindGroup instead of the derive
+// The two textures have the same role and will be swapped in the bind group
+// In the layout, one is `ReadOnly`, the other is `WriteOnly`
+impl AsBindGroup for GameOfLifeImages {
+    type Data = ();
+
+    fn label() -> Option<&'static str> {
+        Some("GameOfLifeImages")
+    }
+
+    fn unprepared_bind_group(
+        &self,
+        _layout: &BindGroupLayout,
+        _render_device: &RenderDevice,
+        _images: &RenderAssets<Image>,
+        _fallback_image: &bevy::render::texture::FallbackImage,
+    ) -> Result<UnpreparedBindGroup<Self::Data>, AsBindGroupError> {
+        Ok(UnpreparedBindGroup {
+            bindings: vec![],
+            data: (),
+        })
+    }
+
+    fn bind_group_layout_entries(_render_device: &RenderDevice) -> Vec<BindGroupLayoutEntry>
+    where
+        Self: Sized,
+    {
+        vec![
+            BindGroupLayoutEntry {
+                binding: 0,
+                visibility: ShaderStages::COMPUTE,
+                ty: BindingType::StorageTexture {
+                    access: StorageTextureAccess::ReadOnly,
+                    format: TextureFormat::R32Float,
+                    view_dimension: TextureViewDimension::D2,
+                },
+                count: None,
+            },
+            BindGroupLayoutEntry {
+                binding: 1,
+                visibility: ShaderStages::COMPUTE,
+                ty: BindingType::StorageTexture {
+                    access: StorageTextureAccess::WriteOnly,
+                    format: TextureFormat::R32Float,
+                    view_dimension: TextureViewDimension::D2,
+                },
+                count: None,
+            },
+        ]
```

Comment:
```suggestion
        vec![
            texture_storage_2d(TextureFormat::R32Float, StorageTextureAccess::ReadOnly)
                .build(0, ShaderStages::COMPUTE),
            texture_storage_2d(TextureFormat::R32Float, StorageTextureAccess::WriteOnly)
                .build(1, ShaderStages::COMPUTE),
        ]
```

You could replace this with those helpers from `BindGroupLayoutEntries`. I just noticed it's not possible to use the full entries api here because you can't get the actual vec of entries easily, so I'll need to fix that.